core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/aggregate/DistributedAggregation.scala

```diff
@@ -32,8 +32,8 @@ case class DistributedAggregation[P <: AnyRef](
     iterate: (P, Tuple) => P,
     // PartialObject + PartialObject => PartialObject
     merge: (P, P) => P,
-    // PartialObject => FinalObject
-    finalAgg: P => Tuple,
+    // PartialObject + Tuple Builder => Tuple Builder
+    finalAgg: (P, Tuple.BuilderV2) => Tuple.BuilderV2,
```

Comment:
I suggest we keep the original API `P => Tuple`. When write an aggregate function, we can return a tuple with a single column, which is the result. Then in the `FinalAggregateOpExec`, we combine them into a new tuple